client/my-sites/checkout/controller.jsx

```diff
@@ -160,6 +160,7 @@ export function upsellNudge( context, next ) {
 			shouldShowCart={ false }
 			clearTransaction={ true }
 			purchaseId={ Number( receiptId ) }
+			isComingFromSignup={ !! context.query.signup }
```

Comment:
Unrelated to this fix - can we use this prop to log purchases from Signup (vs. purchases from later in the user experience)?

Reply:
I don't think so. This prop will also be true when going through the site launch flow for a private site, and also when going through the flow via "Add new site". So, purchases made here will not reflect a signup purchase.

Reply:
If we want to include post-checkout upsell purchases(Quick Start session/GSuite) as purchases from Signup, then we need to carry over the prop to the upsell nudge URLs. We would still have the issue of having to disambiguate purchases made during the launch flow.

Reply:
> This prop will also be true when going through the site launch flow for a private site, and also when going through the flow via "Add new site". So, purchases made here will not reflect a signup purchase.

I would consider 'going through the flow via "Add new site"' as a Signup purchase (a user is creating a new site), but you're right about the launch flow purchases - those are definitely different.

Thanks for the explanation!